web/src/pages/MainPage.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { FC, useEffect, useMemo, useRef, useState } from 'react';
+import styled from 'styled-components';
+import LiveRoomList from '../components/LiveRoomList';
+import {
+  Button,
+  Divider,
+  FormControlLabel,
+  InputAdornment,
+  Switch,
+  TextField,
+} from '@material-ui/core';
+import Search from '@material-ui/icons/Search';
+import { LiveRoom, ClosedRoom } from '../models';
+import { request } from '../helpers';
+import { useRoomsDispatch, useRoomsState } from '../contexts/RoomsContext';
+import CreateRoomModal from '../components/CreateRoomModal';
+import ClosedRoomList from '../components/ClosedRoomList';
+import { useCallback } from 'react';
+
+const MainPageBlock = styled.div`
+  width: 50%;
+  height: 100%;
+
+  background: white;
+
+  margin: 0 auto;
+  margin-top: 64px;
+  border-radius: 10px;
+
+  padding-left: 16px;
+  padding-right: 16px;
+  padding-bottom: 256px;
+`;
+
+const HeaderBlock = styled.div`
+  padding-top: 8px;
+`;
+
+const HeaderBlockRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const MainPage: FC = () => {
+  const [isLive, setIsLive] = useState(true);
+  const [isCreate, setIsCreate] = useState(false);
+  const [searchInputs, setSearchInputs] = useState('');
+
+  const roomsState = useRoomsState();
+  const roomsDispatch = useRoomsDispatch();
+
+  const isCancelled = useRef(false);
+  useEffect(() => {
+    if (!isCancelled.current) {
+      request
+        .get<{ liveRooms: LiveRoom[]; closedRooms: ClosedRoom[] }>('/rooms')
+        .then((res) =>
+          roomsDispatch({
+            type: 'INIT_ROOMS',
+            liveRooms: res.liveRooms,
+            closedRooms: res.closedRooms,
+          }),
+        )
+        .catch((e) => console.error(e));
+    }
+
+    return () => {
+      isCancelled.current = true;
+      roomsDispatch({ type: 'INIT_ROOMS', liveRooms: [], closedRooms: [] });
+    };
+  }, [roomsDispatch]);
```

Comment:
[Major]
타이밍에 따라 취소되어야 하는 것이 Promise의 콜백, 즉 `roomsDispatch`입니다. 지금의 코드는 아무효과가 없을거에요. then 내부의 작업이 실행되지 않게 해주시는 것이 중요합니다.

[Minor]
Hook 내부에서 한 사이클에 대해서만 유효하면 되는 변수이므로 `useRef`로 선언할 필요없이 내부에 선언하는 것이 1) 다른 코드로 인한 변화를 원천적으로 막을 수 있어 안전하고, 2) 사용되는 스코프가 직관적으로 드러나기 때문에 더 좋습니다~!

```suggestion
  useEffect(() => {
    let isCancelled = false; // Minor feedback
    request
      .get<{ liveRooms: LiveRoom[]; closedRooms: ClosedRoom[] }>('/rooms')
      .then((res) =>
        if (isCancelled) return; // Major feedback
        roomsDispatch({
          type: 'INIT_ROOMS',
          liveRooms: res.liveRooms,
          closedRooms: res.closedRooms,
        }),
      )
      .catch((e) => console.error(e));

    return () => {
      isCancelled = true;
      roomsDispatch({ type: 'INIT_ROOMS', liveRooms: [], closedRooms: [] });
    };
  }, [roomsDispatch]);
```

Reply:
적용 완료